<launch>

    <!-- You should edit these parameters to point to your topics -->

    <arg name="camera_topic" default="/camera_front/image_color" doc="Camera topic name" />
    <arg name="cmd_vel_pub" default="/nav/cmd_vel" doc="The topic that cmd_vel messages will be published on when replaying a map (can be the same as cmd_vel_sub)" />
    <arg name="cmd_vel_sub" default="/husky_velocity_controller/cmd_vel" doc="The topic that cmd_vel messages will be recorded from when making a map (can be the same as cmd_vel_pub)" />
    <arg name="odom_topic" default="/husky_velocity_controller/odom" doc="The topic that odometry information will be read from" />

    <!-- You should not need to edit anything below this line. -->

    <group ns="bearnav2">
        
        <!-- calculates which way to steer from images -->
        <node pkg="bearnav2" type="alignment-ros-1.py" name="alignment" respawn="false" output="screen" args="$(find bearnav2)/src/alignment/config.yaml -OO" />

        <!-- responsible for calculating how far the robot has travelled -->
        <node pkg="bearnav2" type="distance-ros-1.py" name="distance" respawn="true" args="-OO">
            <param name="cmd_vel_topic" type="string" value="$(arg cmd_vel_sub)"/>
            <param name="odom_topic" type="string" value="$(arg odom_topic)"/>
        </node>
        
        <!-- node for creating maps -->
        <node pkg="bearnav2" type="mapmaker-ros-1.py" name="mapmaker" respawn="true" args="-OO">
            <param name="camera_topic" type="string" value="$(arg camera_topic)"/>
            <param name="cmd_vel_topic" type="string" value="$(arg cmd_vel_sub)"/>
        </node>
        
        <!-- node for repeating maps -->
        <node pkg="bearnav2" type="repeater-ros-1.py" name="repeater" respawn="true" args="-OO">
            <param name="camera_topic" type="string" value="$(arg camera_topic)"/>
        </node>
        
        <!-- actually does the talking to the robot to make it move -->
        <node pkg="bearnav2" type="navigator-ros-1.py" name="navigator" respawn="true" args="-OO">
            <param name="cmd_vel_topic" type="string" value="$(arg cmd_vel_pub)"/>
        </node>
        
        <!-- image preprocessing -->
        <node pkg="bearnav2" type="preprocess-ros-1.py" name="preprocess" respawn="true" args="-OO" />

    </group>
</launch>
